src/java.base/share/classes/java/util/ArrayList.java

```diff
@@ -334,6 +361,33 @@ int lastIndexOfRange(Object o, int start, int end) {
         return -1;
     }
 
+    /**
+     * Returns the index of the last occurrence of a matching element
+     * in this list, or -1 if this list does not contain a matching element.
+     * More formally, returns the highest index {@code i} such that
+     * {@code filter.test(get(i))},
+     * or -1 if there is no such index.
+     *
+     * @param filter a predicate to search matching element for
+     * @return the index of the last occurrence of the specified element in
+     *         this list, or -1 if this list does not contain the element
+     * @throws NullPointerException if passed filter is null
+     */
+    public int findLastIndex(Predicate<? super E> filter) {
+        return findLastIndexInRange(filter, 0, size);
+    }
+
+    int findLastIndexInRange(Predicate<? super E> filter, int start, int end) {
```

Comment:
Yeah i though about it but indexOfRange arent private here so either there is a point in it or its just legacy without any particular meaning

Reply:
It is legacy to avoid bridge generation from the SubList.

Before introduction of nestmates in JDK 11, private methods and fields called by inner classes should be declared package-private, as javac has to generate bridge methods at each call site to abide to JVM rules (inner classes are just another class in the package so couldn't call private methods).